public/scripts/friendlycode.js

```diff
@@ -12,103 +8,23 @@
   makeDetails = JSON.parse(decodeURIComponent(makeDetails));
 
   /**
-   * ...
-   */
-  define("hackpub", ["jquery", "selectize"], function($, selectize) {
-
-    // set up CSRF handling
-    var csrf_token = $("meta[name='csrf-token']").attr("content");
-    $("#supportedLocales").selectize();
-
-    return function Hackpub(options) {
-      return {
-        loadCode: function(path, cb) {
-          var url = path;
-          $.ajax({
-            type: "GET",
-            url: url,
-            dataType: 'text',
-            error: function(req) {
-              cb(req);
-            },
-            success: function(html) {
-              cb(null, html, url);
-            }
-          });
-        },
-        saveCode: function(data, originalURL, cb) {
-          $.ajax({
-            type: "POST",
-            url: options.hackpubURL + "/" + localeInfo_lang + "/publish",
-            data: {
-              'html': data.html,
-              'proxied': data.proxied,
-              'metaData': data.metaData,
-              'published': data.published,
-              'pageOperation': $("meta[name='thimble-operation']").attr("content"),
-              'origin': $("meta[name='thimble-project-origin']").attr("content")
-            },
-            dataType: 'json',
-            beforeSend: function(request) {
-              request.setRequestHeader('X-CSRF-Token', csrf_token); // express.js uses a non-standard name for csrf-token
-            },
-            error: function(req) {
-              cb(req);
-            },
-            success: function(result) {
-              var origin = $("meta[name='thimble-project-origin']").attr("content");
-              cb(null, {path: result['remix-id'], url: result['published-url']});
-              // If we were a remix, update the new page's origin.
-              if ($("meta[name='thimble-operation']").attr("content") === "remix" || !origin) {
-                origin = result['remix-id'];
-                $("meta[name='thimble-project-origin']").attr("content",origin);
-              }
-              $("meta[name='thimble-operation']").attr("content","edit");
-
-              // Disable the accidental page-navigation protection
-              // right after a publish:
-              data.dataProtector.disableDataProtection();
-            }
-          });
-        }
-      };
-    };
-  });
-
-  /**
    * This code sets up the various Friendlycode values for
    * publication and remixing.
    */
   define("thimblePage",
-         ["jquery", "friendlycode", "hackpub", "languages"],
-         function($, FriendlycodeEditor, Hackpub, Languages) {
-
-    var makeUrl = document.getElementById("friendly-code").getAttribute("data-make-url"),
-        makeEndpoint = document.getElementById("friendly-code").getAttribute("data-make-endpoint");
+         ["jquery", "friendlycode", "languages"],
+         // XXXBramble - nuke Languages when possible
```

Comment:
Poorly labeled - should have been `XXXSedge` since it's a note to myself. This patch is still WIP